tests/eth-tx-auth.test.ts

```diff
@@ -0,0 +1,607 @@
+      CallData.compile({
+        target: space.address,
+        author: proposal.author,
+        metadataURI: proposal.metadataUri,
+        executionStrategy: proposal.executionStrategy,
+        userProposalValidationParams: proposal.userProposalValidationParams,
+      }),
+      { rawInput: true },
+    );
+
+    // Attempting to replay the proposal creation commit
+    try {
+      await account.invoke(
+        ethTxAuthenticator,
+        'authenticate_propose',
+        CallData.compile({
+          target: space.address,
+          author: proposal.author,
+          metadataURI: proposal.metadataUri,
+          executionStrategy: proposal.executionStrategy,
+          userProposalValidationParams: proposal.userProposalValidationParams,
+        }),
+        { rawInput: true },
+      );
+      expect.fail('Should have failed');
+    } catch (err: any) {
+      expect(err.message).to.contain(shortString.encodeShortString('Commit not found'));
+    }
+  }, 1000000);
+
+  it('a commit cannot be overwritten by a different sender', async () => {
+    await starknet.devnet.restart();
+    await starknet.devnet.load('./dump.pkl');
+    await starknet.devnet.increaseTime(10);
+    await starknet.devnet.loadL1MessagingContract(eth_network, mockStarknetMessaging.address);
+
+    const proposal = {
+      author: signer.address,
+      metadataUri: ['0x1', '0x2', '0x3', '0x4'],
+      executionStrategy: {
+        address: '0x0000000000000000000000000000000000005678',
+        params: ['0x0'],
+      },
+      userProposalValidationParams: [
+        '0xffffffffffffffffffffffffffffffffffffffffff',
+        '0x1234',
+        '0x5678',
+        '0x9abc',
+      ],
+    };
+
+    const proposeCommitPreImage = CallData.compile({
+      target: space.address,
+      selector: selector.getSelectorFromName('propose'),
+      ...proposal,
+    });
+
+    // Commit hash of payload to the Starknet Commit L1 contract
+    const commit = `0x${poseidonHashMany(proposeCommitPreImage.map((v) => BigInt(v))).toString(
+      16,
+    )}`;
+
+    await starknetCommit.commit(ethTxAuthenticator.address, commit, { value: 18485000000000 });
+
+    // Committing the same commit but different sender
+    await starknetCommit
+      .connect(invalidSigner)
+      .commit(ethTxAuthenticator.address, commit, { value: 18485000000000 });
+
+    // Checking that both L1 -> L2 messages has been propogated
+    expect((await starknet.devnet.flush()).consumed_messages.from_l1).to.have.a.lengthOf(2);
+
+    // The commit by the correct signer should be accepted
+    await account.invoke(
+      ethTxAuthenticator,
+      'authenticate_propose',
+      CallData.compile({
+        target: space.address,
+        author: proposal.author,
+        metadataURI: proposal.metadataUri,
+        executionStrategy: proposal.executionStrategy,
+        userProposalValidationParams: proposal.userProposalValidationParams,
+      }),
+      { rawInput: true },
+    );
+  }, 1000000);
+});
```

Comment:
can now add a test for https://github.com/snapshot-labs/sx-starknet/issues/555 :D (probably best in a different PR :p)